src/app/pipes/alm-valid-link-types.pipe.ts

```diff
@@ -0,0 +1,25 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { WorkItem } from '../models/work-item';
+
+@Pipe({ name: 'almValidLinkTypes', pure: true })
+export class AlmValidLinkTypes implements PipeTransform {
+  transform(arr: any[], workItem: WorkItem): any {
+    let opLinkTypes: any[] = [];
+    arr.forEach((item) => {
+      if (item.relationships.source_type.data.id == workItem.relationships.baseType.data.id) {
+        opLinkTypes.push({
+          name: item.attributes['forward_name'],
+          linkId: item.id,
+          linkType: 'forward'
+        });
+      } else if (item.relationships.target_type.data.id == workItem.relationships.baseType.data.id){
```

Comment:
If send a buggy link, it will not blow up, it'll just not show up the link in the list. If none of these conditions passes the link will not be pushed into the list.